11-functions.Rmd

```diff
@@ -5,6 +5,123 @@
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21a84b32106_0_53")
 ```
+
+### Recognize when to write a custom function
+
+There's a lot of reasons writing your own custom functions might be a thing you need to do.
+
+#### Cutting down on repetitive code
+
+We've discussed how DRY code is easier to maintain and understand! It is easier to fix code in one place as opposed to three or more places. A general guideline is that if you need two use a same chunk of code more than twice you probably should invest the time to make a custom function for it.
+
+```{r, fig.align='center', fig.alt = "The parrot is saying 'I’m using this code chunk three times. Updating this might get hairy… Good time to make a custom function!' The computer has three identical chunks of code on the screen.
+", out.width="100%", echo = FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g341450bc187_100_5")
+```
+
+#### More readable code
+
+Custom functions can allow you to organize code into manageable chunks.
+By creating custom functions it can be easier to organize our code in a way thats more readable.
+
+
+### Writing a function
+
+A common way that a custom function might come about is:
+
+1. You've written some initial code that takes some object as input.
+2. You now realize you will need to run this series of code more than twice.
+
+
+#### Starting a new function
+
+To get started with writing a new function, you can follow these steps to make a custom function:
+
+1. Open up a new R script file to copy and paste our basic template we have below.
+```
+#' Title here
+#'
+#' @description This is what the function does
+#'
+#' @param arg1 A first argument
+#' @param arg2 A second argument
+#'
+#' @return What is returned
+#' @export
+#'
+#' @examples
+#'
+#' function(arg1, arg2)
+#'
+#'
+function_name <- function(arg1, arg2) {
+
+    # Your main code here
+
+    return(output)
+}
+```
+Your basic custom function is going to follow this type of structure:
+
+We will walk through how to use this template here.
+
+This top part of the template is the documentation. It's formatted in a way that would
+allow you to incorporate it into a package if you would like.
+
+2. It's likely you might already have some code drafted that has some of the code you need for your function. Paste this code in the part of this template that says `# Your main code here`.
+3.  Let's write a `@description` field. Describe the goal of the function.
+4. Next you'll likely want to sculpt your code, carefully thinking about its usage  and what types of options may need to be controlled by those who will use this function.
```

Comment:
```suggestion
4. Next you'll likely want to sculpt your code, carefully thinking about its usage  and what types of options may need to be controlled by those who will use this function. This may require some adjustments to arguments and default values.
```

Reply:
adding some more details